platform/mellanox/hw-management.mk

```diff
@@ -24,3 +24,117 @@ MLNX_HW_MANAGEMENT = hw-management_1.mlnx.$(MLNX_HW_MANAGEMENT_VERSION)_$(CONFIG
+							--build_root $(BUILD_WORKDIR) $(LOG_SIMPLE)
+
+	$(BUILD_WORKDIR)/$($(MLNX_HW_MANAGEMENT)_SRC_PATH)/hw-mgmt/recipes-kernel/linux/deploy_kernel_patches.py \
+							--dst_accepted_folder $(PTCH_DIR) \
+							--dst_candidate_folder $(NON_UP_PTCH_DIR) \
+							--series_file $(PTCH_LIST) \
+							--config_file $(KCFG_LIST) \
+							--kernel_version $(KERNEL_VERSION) \
+							--os_type sonic $(LOG_SIMPLE)
+
+	# Post-processing
+	integration-scripts/hwmgmt_kernel_patches.py post \
+							--patches $(PTCH_DIR) \
+							--non_up_patches $(NON_UP_PTCH_DIR) \
+							--config_inclusion $(KCFG_LIST) \
+							--series $(PTCH_LIST) \
+							--current_non_up_patches $(HWMGMT_NONUP_LIST) \
+							--build_root $(BUILD_WORKDIR) $(LOG_SIMPLE)
+	
+	# Commit the changes in linux kernel and and log the diff
+	pushd $(BUILD_WORKDIR)/src/sonic-linux-kernel
+	git add -- patch/; git commit -m "Intgerate HW-MGMT ${MLNX_HW_MANAGEMENT_VERSION} Changes";
+
+	echo -en "\n###-> series file changes in sonic-linux-kernel <-###\n" >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD -- patch/series >> ${USER_OUTFILE}
+
+	echo -en "\n###-> kconfig-inclusions file changes in sonic-linux-kernel <-###\n" >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD -- patch/kconfig-inclusions >> ${USER_OUTFILE}
+
+	echo -en "\n###-> kconfig-exclusions file changes in sonic-linux-kernel <-###\n" >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD -- patch/kconfig-exclusions >> ${USER_OUTFILE}
+
+	echo -en '\n###-> Summary of files updated in sonic-linux-kernel <-###\n' >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD --stat --output=${TMPFILE_OUT}
+	cat ${TMPFILE_OUT} | tee -a ${USER_OUTFILE}
+	popd
+
+	# Commit the changes in buildimage and log the diff
+	pushd $(BUILD_WORKDIR)
+	git add -- $($(MLNX_HW_MANAGEMENT)_SRC_PATH)
+	git add -- $(PLATFORM_PATH)/non-upstream-patches/
+	git add -- $(PLATFORM_PATH)/hw-management.mk
+	git commit -m "Intgerate HW-MGMT ${MLNX_HW_MANAGEMENT_VERSION} Changes";
+
+	echo -en '\n###-> Non Upstream series.patch changes <-###\n' >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD -- $(PLATFORM_PATH)/non-upstream-patches/series.patch >> ${USER_OUTFILE}
+
+	echo -en '\n###-> Non Upstream patch list file <-###\n' >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD -- $($(MLNX_HW_MANAGEMENT)_SRC_PATH)/hwmgmt_nonup_patches >> ${USER_OUTFILE}
+
+	echo -en '\n###-> hw-mgmt submodule update <-###\n' >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD -- $($(MLNX_HW_MANAGEMENT)_SRC_PATH)/hw-mgmt >> ${USER_OUTFILE}
+
+	echo -en '\n###-> hw-management make file version change <-###\n' >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD -- $(PLATFORM_PATH)/hw-management.mk >> ${USER_OUTFILE}
+	
+	echo -en '\n###-> Summary of buildimage changes <-###\n' >> ${USER_OUTFILE}
+	git diff --no-color  HEAD~1 HEAD --stat --output=${TMPFILE_OUT} -- $(PLATFORM_PATH)
+	cat ${TMPFILE_OUT} | tee -a ${USER_OUTFILE}
+	popd
```

Comment:
Should we do git push if there is CREATE_BRANCH?

Reply:
We don't know the remote to push. User should do this manually as of now